### PR TITLE
Update Factoring JN to use dimod gate generators

### PR DESCRIPTION
**Pull request by JoelPasvolsky**
Needs the SDK that incorporates a dimod release with https://github.com/dwavesystems/dimod/pull/1036

### CODE REVIEW

File: 01-factoring-overview.ipynb [hunk 1]
```diff
@@ -631,7 +593,7 @@
    "name": "python",
    "nbconvert_exporter": "python",
    "pygments_lexer": "ipython3",
-   "version": "3.7.0"
```
Comment (hhtong) [inline 01-factoring-overview.ipynb:634]: Not sure if this was intentional, but why change the python version?

**Review by hhtong — APPROVED** [review]
LGTM!

File: 01-factoring-overview.ipynb (hunk 1 above)
Comment (JoelPasvolsky) [reply 01-factoring-overview.ipynb:634]: JN just records the last version of Python in its run env

**Comment by mhlr** [comment]
Is there still a reason for not merging? Gate generators are merged into dimod now.

**Comment by JoelPasvolsky** [comment]
> Is there still a reason for not merging? Gate generators are merged into dimod now.

I forgot it was here. I'll update the requirements file to the latest sdk and it's good to merge.

File: requirements.txt [hunk 2]
```diff
@@ -1,4 +1,4 @@
-dwave-ocean-sdk>4.2.0
+dwave-ocean-sdk>4.3.0
```
Comment (hhtong) [inline requirements.txt]: Ah, this should be: `>=4.3.0`

Comment (JoelPasvolsky) [reply requirements.txt]: Yes, I saw that and force pushed it already

**Pull request merged**